include/alpaka/rand/RandDefault.hpp

```diff
@@ -0,0 +1,243 @@
+                        for(unsigned int a = 0; a < N; ++a)
+                        {
+                            ret
+                                ^= (static_cast<T>(engine())
+                                    << (sizeof(BitsT) * std::numeric_limits<unsigned char>::digits * a));
+                        }
+                        return ret;
+                    }
+                };
+
+                //! The GPU random number uniform distribution.
+                template<typename T>
+                class UniformReal
+                {
+                    static_assert(
+                        std::is_floating_point<T>::value,
+                        "Return type of UniformReal must be floating point.");
+
+                    using BitsT = typename detail::BitsType<T>::type;
+
+                public:
+                    UniformReal() = default;
+
+                    template<typename TEngine>
+                    ALPAKA_FN_HOST_ACC auto operator()(TEngine& engine) -> T
+                    {
+                        constexpr BitsT limit = static_cast<BitsT>(1) << std::numeric_limits<T>::digits;
+                        const BitsT b = UniformUint<BitsT>()(engine);
+                        const auto ret = static_cast<T>(b & (limit - 1)) / limit;
+                        return ret;
+                    }
+                };
+
+                /*! The GPU random number normal distribution.
+                 *
+                 * \note
+                 * This type contains state and is not thread-safe: To be used
+                 * per thread, not shared.
+                 */
+                template<typename Acc, typename T>
+                class NormalReal
+                {
+                    static_assert(
+                        std::is_floating_point<T>::value,
+                        "Return type of NormalReal must be floating point.");
+
+                    const Acc& m_acc;
+                    T m_buf = std::numeric_limits<T>::quiet_NaN();
+
+                public:
+                    /*! \warning Retains a reference to \p acc, thus must not
+                     * outlive it.
+                     */
+                    NormalReal(const Acc& acc) : m_acc(acc)
+                    {
+                    }
+
+                    //! The copy ctor does not copy `m_buf`.
+                    NormalReal(const NormalReal& o) : m_acc(o.m_acc)
+                    {
+                    }
+                    //! The move ctor clears `m_buf` of source.
+                    NormalReal(NormalReal&& o) : m_acc(o.m_acc), m_buf(o.m_buf)
+                    {
+                        o.m_buf = std::numeric_limits<T>::quiet_NaN();
+                    }
+
+                    //! The copy must not obtain a copy of `m_buf`.
+                    NormalReal& operator=(const NormalReal& o)
+                    {
+                        ALPAKA_ASSERT_OFFLOAD(&m_acc == &o.m_acc);
+                    }
```

Comment:
This does nothing. Basically copy should be deleted because we cannot have two distributions with the same cached value around.